### DISCUSSION TITLE
サンプル（helloev3）ビルド時のエラーについて

### DISCUSSION THREAD

**Discussion by WatanabeMasato** [toppers/users-forum#112] [ET-Robocon]
はじめまして、Mattと申します。この度EV３をETロボコン向けTOPPERSを使用したいと思い“#2 Windows(WSL)版開発環境構築【後輩たちのためのEV3rt講座(改)】”を参考にして環境を構築しました。環境は下記のとおりです。
ＴｈｉｎｋＰＡＤL15Gen3
WINDOWS11　home
WSL 1
サンプルをビルドすると下記のエラーが表示されます。解決策をお教え願えないでしょうか
よろしくお願いします
matt@LAPTOP-GGU2I405:/mnt/c/ev3rt-1.1-release/hrp3/sdk/workspace$ make app=helloev3
rm -rf /mnt/c/ev3rt-1.1-release/hrp3/sdk/workspace/.././OBJ/
Traceback (most recent call last):
        2: from /mnt/c/ev3rt-1.1-release/hrp3/sdk/workspace/../../configure.rb:47:in `<main>'
        1: from /home/matt/.rbenv/versions/2.7.0/lib/ruby/2.7.0/rubygems/core_ext/kernel_require.rb:92:in `require'
/home/matt/.rbenv/versions/2.7.0/lib/ruby/2.7.0/rubygems/core_ext/kernel_require.rb:92:in `require': cannot load such file -- shell (LoadError)
make: *** [../common/Makefile.workspace:152: prepare-obj-folder] Error 1

**Comment by Y-Ishioka** [comment]
正式な回答はTOPPERS prjの方々から出ると思いますが、コメントしておきます。
昔、TOPPERS/ASP3をビルドしようとしたときに configureの実行で似たエラーが出たような気がします。
Ruby 2.7.0では標準で shellライブラリがないため、その時は手動でロードして問題を回避しました。
以下のようなコマンドを実行したようです。

sudo gem install shell

PRを兼ねて、Qiitaのページを付けておきます。
https://qiita.com/Yukiya_Ishioka/items/2317b35ab0b98e15f6d0

**Comment by WatanabeMasato** [comment]
早々の返信ありがとうございます
月曜日試してみます
今後ともよろしくお願いします

2023年4月28日(金) 12:50 Y-Ishioka ***@***.***>:
